include/zephyr/gnss/rtk/decoder.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2025 Croxel Inc.
+ * Copyright (c) 2025 CogniPilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_GNSS_RTK_DECODER_H_
+#define ZEPHYR_INCLUDE_GNSS_RTK_DECODER_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+/**
+ * @brief Get an RTK frame from buffer
+ *
+ * Used by RTK clients to extract frames from a data-buffer.
+ *
+ * @param[in] buf Buffer holding encoded data.
+ * @param[in] buf_len Buffer length.
+ * @param[out] data Pointer to the decoded frame.
+ * @param[out] data_len Length of the decoded frame
+ *
+ * @return Zero if successful.
+ * @return -ENOENT if no frames have been decoded successfully.
+ * @return -EAGAIN if there's an incomplete frame starting at the data pointer.
+ * @return Other negative error code if decoding failed.
+ */
+int gnss_rtk_decoder_frame_get(uint8_t *buf, size_t buf_len,
```

Comment:
```suggestion
int gnss_rtk_decoder_frame_get(const uint8_t *buf, size_t buf_len,
```

Reply:
This was intentionally left as a mutable array because I was considering there may be some encoded frames with byte stuffing, hence leaving the option for the API to do things behind this API call.

It is not certainly the case with the only one implementation (RTCM3), but I'd prefer keep it this way unless you feel strongly about enforcing it until an implementation that requires it comes by.

Reply:
I guess I am just confused by the fact of having two functions named exactly the same but having [different signatures](https://github.com/zephyrproject-rtos/zephyr/pull/90890/files#diff-bb1190f6232750a33cab16130e770d0dd243fdea770932b3a753e016b5eebd4dR30-R31)

Reply:
Should be consistent now. Removed stray header file under `subsys/gnss/rtk/decoder.h`